### PR TITLE
Implemented the vander and test

### PR DESCRIPTION
**Pull request by Fathi0011**
Close https://github.com/unifyai/ivy/issues/9553

### CODE REVIEW

**Comment by Fathi0011** [comment]
Close #9553

File: .idea/ivy.iml [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
   </component>
   <component name="PyDocumentationSettings">
+    <option name="format" value="NUMPY" />
     <option name="myDocStringFormat" value="NumPy" />
   </component>
   <component name="TestRunnerService">
-    <option name="projectConfiguration" value="pytest" />
     <option name="PROJECT_TEST_RUNNER" value="py.test" />
   </component>
 </module>
```
Comment (guillesanbri) [inline .idea/ivy.iml:15]: this file should be removed from the commit :slightly_smiling_face:

File: .idea/misc.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
   <component name="JavaScriptSettings">
     <option name="languageLevel" value="ES6" />
   </component>
-  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.8" project-jdk-type="Python SDK" />
+  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.11 (ivy_dev_) (3)" project-jdk-type="Python SDK" />
   <component name="PyPackaging">
     <option name="earlyReleasesAsUpgrades" value="true" />
   </component>
```
Comment (guillesanbri) [inline .idea/misc.xml:9]: same as above, project files should not be modified

File: ivy/functional/frontends/torch/linalg.py [hunk 3]
```diff
@@ -170,3 +170,7 @@ def eig(input, *, out=None):
 @with_unsupported_dtypes({"1.11.0 and below": ("bfloat16", "float16")}, "torch")
 def solve(input, other, *, out=None):
     return ivy.solve(input, other, out=out)
+ 
+@to_ivy_arrays_and_back
+def vander(A, *, increasing=False, out=None):
```
Comment (guillesanbri) [inline ivy/functional/frontends/torch/linalg.py:175]: variable names and keywords of the functions should match those in torch, i.e. `x`, `N=None`, `increasing=False`

File: ivy/functional/frontends/torch/linalg.py [hunk 4]
```diff
@@ -170,3 +170,7 @@ def eig(input, *, out=None):
...
+    return ivy.vander(A, out=out)
```
Comment (guillesanbri) [inline ivy/functional/frontends/torch/linalg.py:176]: both kwargs (`N` and `increasing`) should be dealt with

File: ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py [hunk 5]
```diff
@@ -880,3 +880,32 @@ def test_torch_tensorinv(
...
+    helpers.test_frontend_function(
+        input_dtypes=dtype,
+        as_variable_flags=as_variable,
+        with_out=with_out,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        A=x,
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:910]: should include all args :slightly_smiling_face:

File: ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py [hunk 6]
```diff
@@ -880,3 +880,32 @@ def test_torch_tensorinv(
         input=x,
         ind=ind,
     )
+
+ # vander
+@handle_frontend_test(
+    fn_tree="torch.linalg.vander",
+    dtype_and_x=_get_dtype_and_square_matrix(),
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:887]: seems like vander input is a 1D array

**Review by guillesanbri — CHANGES_REQUESTED** [review]
Hi! thank you very much for the contribution :smile: I have left some comments on the PR :slightly_smiling_face:

File: ivy/functional/frontends/torch/linalg.py (hunk 4 above)
Comment (Fathi0011) [reply ivy/functional/frontends/torch/linalg.py:176]: so it should be this ? 

def vander(x, *, increasing=False, N=None):
    return ivy.vander(x, increasing=increasing, out=N)

Comment (guillesanbri) [reply ivy/functional/frontends/torch/linalg.py:176]: You can check the docs here: https://pytorch.org/docs/1.11/generated/torch.vander.html?highlight=vander#torch.vander

the order of the kwargs should be the same, and you should deal with any arguments that are not matched in the `ivy` function in the frontend function (which I guess it's the case with `N`) :slightly_smiling_face:

**Comment by guillesanbri** [comment]
keep in mind that the project files have to be removed from the commit, which is not the same as deleting the files :slightly_smiling_face:

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**